packages/flutter_test/lib/src/matchers.dart

```diff
@@ -2490,7 +2498,16 @@ class _MatchesSemanticsData extends Matcher {
       return failWithDescription(matchState, 'No SemanticsData provided. '
         'Maybe you forgot to enable semantics?');
     }
-    final SemanticsData data = node is SemanticsNode ? node.getSemanticsData() : (node as SemanticsData);
+
+    late final SemanticsData data;
```

Comment:
nit: you should be able to drop the `late` here. Dart is smart enough that it shouldn't need it here.

Reply:
Huh, neat! Didn't know that one!